src/main/java/org/medicmobile/webapp/mobile/RDToolkitSupport.java

```diff
@@ -0,0 +1,219 @@
+
+		log(this, "RDToolkitSupport :: process requestCode=%s", requestCode);
+
+		switch (requestCode) {
+			case RDTOOLKIT_PROVISION_ACTIVITY_REQUEST_CODE:
+				if (resultCode != RESULT_OK) {
+					throw new RuntimeException("RDToolkitSupport :: Bad result code for the provisioned RD Test: " + resultCode);
+				}
+
+				try {
+					JSONObject response = parseProvisionTestResponseToJson(intentData);
+					return makeProvisionTestJavaScript(response);
+
+				} catch (Exception exception) {
+					error(exception, "RDToolkitSupport :: Problem serialising the provisioned RD Test");
+					return safeFormat("console.log('Problem serialising the provisioned RD Test: %s')", exception);
+				}
+
+			case RDTOOLKIT_CAPTURE_ACTIVITY_REQUEST_CODE:
+				if (resultCode != RESULT_OK) {
+					throw new RuntimeException("RDToolkitSupport :: Bad result code for capturing result: " + resultCode);
+				}
+
+				try {
+					JSONObject response = parseCaptureResponseToJson(intentData);
+					return makeCaptureResponseJavaScript(response);
+
+				} catch (Exception exception) {
+					error(exception, "RDToolkitSupport :: Problem serialising the captured result");
+					return safeFormat("console.log('Problem serialising the captured result: %s')", exception);
+				}
+
+			default:
+				throw new RuntimeException("RDToolkitSupport :: Bad request type: " + requestCode);
+		}
+	}
+
+	Intent provisionRDTest(String sessionId, String patientName, String patientId, String rdtFilter, String monitorApiURL) {
+		ProvisionMode provisionMode = !rdtFilter.trim().matches("\\S+") ? ProvisionMode.CRITERIA_SET_AND : ProvisionMode.CRITERIA_SET_OR;
+		Intent intent = RdtIntentBuilder
+				.forProvisioning()
+				.setCallingPackage(ctx.getPackageName())
+				.setReturnApplication(ctx)
+				// Type of RD Test to choose from
+				.requestProfileCriteria(rdtFilter, provisionMode)
+				// Unique ID for RD Test
+				.setSessionId(sessionId)
+				// First line text to display in RDToolkit App and to differentiate running tests
+				.setFlavorOne(patientName)
+				// Second line text to display in RDToolkit App and to differentiate running tests
+				.setFlavorTwo(patientId)
+				.setCloudworksBackend(monitorApiURL, patientId)
+				.build();
+
+		if (intent.resolveActivity(ctx.getPackageManager()) != null) {
+			ctx.startActivityForResult(intent, RDTOOLKIT_PROVISION_ACTIVITY_REQUEST_CODE);
+		}
+
+		return intent;
+	}
+
+	Intent captureRDTest(String sessionId) {
+		Intent intent = RdtIntentBuilder
+				.forCapture()
+				// Unique ID for RD Test
+				.setSessionId(sessionId)
+				.build();
+
+		if (intent.resolveActivity(ctx.getPackageManager()) != null) {
+			ctx.startActivityForResult(intent, RDTOOLKIT_CAPTURE_ACTIVITY_REQUEST_CODE);
+		}
+
+		return intent;
+	}
+
+//> PRIVATE HELPERS
+
+	private String makeProvisionTestJavaScript(Object response) {
+		String javaScript = "try {" +
+				"const api = angular.element(document.body).injector().get('AndroidApi');" +
```

Comment:
Are we planning to integrate this into older versions of the app (before 3.10)? 
If not, we can use `window.CHTCore.AndroidApi` instead. The `angular.element...` is deprecated as on 3.11

Reply:
Because we're using Angular 10, I don't think we're going to support 3.10 and before. 
Switching to use: `window.CHTCore.AndroidApi` 💪